.github/workflows/test.yml

```diff
@@ -31,25 +31,29 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-20.04, macos-14]
+        include:
+          - os: ubuntu-20.04
+            protoc: sudo apt install -y protobuf-compiler
+            flags: --all-features
+          - os: macos-14
+            protoc: brew install protobuf
+            flags: --all-features
+          - os: windows-latest
+            protoc: choco install protoc
+            flags: --no-default-features -F extensions
     runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout the repo
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
 
-      - if: ${{ runner.os == 'macOS' }}
-        name: Install protoc
-        run: brew install protobuf
-
-      - if: ${{ runner.os == 'Linux' }}
-        name: Install protoc
-        run: sudo apt install -y protobuf-compiler
+      - name: Install protoc
+        run: ${{ matrix.protoc }}
```

Comment:
I'm not sure I can explain why... But I preferred this better before. Putting the install command in the variables for the matrix just feels weird...

More importantly, the `oldstable` matrix still does it the other way. These should probably match.